File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int>mp;
        int n = nums.size();
            int rem = 0;
        for(int i = 0; i < n; i++){
            int rem = target - nums[i];
            while(mp.find(rem) != mp.end()){
                if(nums[i] + rem == target)
                return {i,mp[rem]};
            }
            mp[nums[i]] = i;
        }
        return {};
    }
};

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer,Integer> mp = new HashMap<>();
        for(int i = 0; i < nums.length; i++){
            mp.put(nums[i], i);
        }
        int rem = 0;
        for(int i =0; i < nums.length; i++){
            rem = target - nums[i];
            if(mp.containsKey(rem)){
                int idx = mp.get(rem);

                if(idx != i){
                    return new int[] {i, idx};
                }
            }
        }
        return new int[] {};
    }
}